repl/format.go

```diff
@@ -52,6 +52,9 @@ func FormatValue(v cty.Value, indent int) string {
 			return strconv.Quote(v.AsString())
 		case cty.Number:
 			bf := v.AsBigFloat()
+			if bf.IsInt() {
+				return bf.String()
+			}
```

Comment:
I guess the result is probably the same for integers in the end anyway, but I think elsewhere we've used the `f` formatting mode which _never_ uses scientific notation. It feels weird to me to use scientific notation in Terraform since we're not typically _doing science_, but I guess it'll probably produce a better result for very small fractional numbers... :thinking: maybe we should (separately from this PR) do a survey of how we've been formatting numbers to strings for the UI elsewhere and decide what our "standard" ought to be.

Reply:
Thanks! After a quick look, I can only find `bf.Text('f', -1)` as the other way we display numbers in Terraform: https://github.com/hashicorp/terraform/blob/5132abf6386830420797c739fb535f5f5b0b424f/command/format/diff.go#L720-L722

That's certainly clearer than my original branching approach, so I've updated the commit to use that. No more scientific notation!

I can't find any test cases that don't work well with this. Even very large integer values print correctly:

```shellsession
$ echo 12345678901234567890123456789012345678901234567890 | terraform console
12345678901234567890123456789012345678901234567890
```